include/dump_utils.hpp

```diff
@@ -0,0 +1,132 @@
+
+/*
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+#pragma once
+
+#include "async_resp.hpp"
+#include "error_messages.hpp"
+#include "logging.hpp"
+#include "utility.hpp"
+
+#include <boost/algorithm/string/case_conv.hpp>
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/shared_ptr.hpp>
+#include <boost/url/url_view.hpp>
+#include <dbus_singleton.hpp>
+#include <dbus_utility.hpp>
+#include <http_stream.hpp>
+#include <ibm/utils.hpp>
+
+#include <cstddef>
+#include <random>
+#include <string>
+
+namespace redfish
+{
+namespace dump_utils
+{
+
+inline void getValidDumpEntryForAttachment(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp, const std::string& url,
+    std::function<void(const std::string& objectPath,
+                       const std::string& entryID,
+                       const std::string& dumpType)>&& callback)
+{
+    std::string dumpType;
+    std::string entryID;
+    std::string entriesPath;
+
+    if (crow::utility::readUrlSegments(url, "redfish", "v1", "Managers", "bmc",
+                                       "LogServices", "Dump", "Entries",
+                                       std::ref(entryID), "attachment"))
+    {
+        // BMC type dump
+        dumpType = "BMC";
+        entriesPath = "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/";
+    }
+    else if (crow::utility::readUrlSegments(
+                 url, "redfish", "v1", "Systems", "system", "LogServices",
+                 "Dump", "Entries", std::ref(entryID), "attachment"))
+    {
+        // System type
+        dumpType = "System";
+        entriesPath = "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/";
+    }
+    if (dumpType.empty() || entryID.empty())
+    {
+        redfish::messages::resourceNotFound(asyncResp->res, "BMC dump",
```

Comment:
Shouldn't this be schema ? Or at least "<dump type> dump"

Reply:
Dump is not in the schema - https://redfish.dmtf.org/schemas/v1/.

However, in dump_offload implementation, both paths are in the routing.

-  "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/<str>/attachment/"
- "/redfish/v1/Systems/system/LogServices/Dump/Entries/<str>/attachment/"

This above checking is to double-check the correctness.

Reply:
I think there is a matching schema for Dump.

./redfish/v1/schema/LogEntry_v1.xml:


-    `<String>/redfish/v1/Managers/{ManagerId}/LogServices/{LogServiceId}/Entries/{LogEntryId}</String>`
-    `<String>/redfish/v1/Systems/{ComputerSystemId}/LogServices/{LogServiceId}/Entries/{LogEntryId}</String>`